src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts

```diff
@@ -150,6 +150,20 @@ export class ExerciseDetailsStudentActionsComponent {
         return participationStatus(this.exercise);
     }
 
+    /**
+     * Display the 'open code editor' or 'clone repo' buttons if
+     * - the participation is initialized (build plan exists, no clean up happened), or
+     * - the participation is inactive (build plan cleaned up), but can not be resumed (e.g. because we're after the due date)
+     */
+    shouldDisplayIDEButtons(): boolean {
+        const status = participationStatus(this.exercise);
+        return (
+            (status === ParticipationStatus.INITIALIZED || (status === ParticipationStatus.INACTIVE && !isStartExerciseAvailable(this.exercise))) &&
+            !!this.exercise.studentParticipations && // This seems weird but silences eslint
```

Comment:
```suggestion
            !!this.exercise.studentParticipations &&
```
This is not weird at all. `studentParticipations` can be undefined. So in case it would be falsy, it would return `undefined` instead of `false`. `!!` ensures that all falsy values return `false` due to the double conversion.

Reply:
(The method requires the return type `boolean`)

Reply:
Hm yeah guess you're right. I think I expected the whole expression, in case one component is falsy, to evaluate to ``false``, but ``&&`` actually preserves the first falsy value and returns it unchanged. Can still learn new things about JS every day :P
![grafik](https://user-images.githubusercontent.com/23171488/151579263-f205e5d7-868f-459b-b33b-d6ab3f5ce934.png)

Reply:
Well, you have to differentiate two things: "==" and "===". The last one is actually a comparison by type. The first one isn't.
And then you have to remember that we don't use JavaScript but TypeScript so if there is a specific return type required you actually need the type not just a soft equivalent.

Reply:
😂 Just wanted to add a meme to the conversation .. :D ofc I know that 🙃

Reply:
haha alright, you never know 😂